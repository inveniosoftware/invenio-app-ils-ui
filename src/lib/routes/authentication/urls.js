import { generatePath } from 'react-router-dom';
const AuthenticationBase = '/';

const AuthenticationRoutesList = {
  login: `${AuthenticationBase}login`,
  redirectUrlAfterLogin: `${AuthenticationBase}login?next=:nextUrl`,
  confirmEmail: `${AuthenticationBase}confirm-email`,
};

const AuthenticationRoutesGenerators = {
  redirectAfterLogin: (nextUrl) =>
    generatePath(AuthenticationRoutesList.redirectUrlAfterLogin, {
      nextUrl: nextUrl,
    }),
};

export const AuthenticationRoutes = {
  ...AuthenticationRoutesList,
  ...AuthenticationRoutesGenerators,
};
