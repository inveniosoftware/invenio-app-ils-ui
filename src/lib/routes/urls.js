export { FrontSiteRoutes } from '@routes/frontsite/frontsiteUrls';
export { BackOfficeRoutes } from '@routes/backoffice/backofficeUrls';
export { AuthenticationRoutes } from './authentication/urls';
export { AcquisitionRoutes } from './backoffice/AcquisitionUrls';
export { ProviderRoutes } from './backoffice/ProvidersUrls';
export { ILLRoutes, DetailsRouteByPidTypeFor } from './backoffice/ILLUrls';
