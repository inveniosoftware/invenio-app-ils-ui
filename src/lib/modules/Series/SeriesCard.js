import { goTo } from '@history';
import LiteratureCover from '@modules/Literature/LiteratureCover';
import LiteratureTitle from '@modules/Literature/LiteratureTitle';
import { SeriesAuthors } from '@modules/Series/SeriesAuthors';
import { FrontSiteRoutes } from '@routes/urls';
import _get from 'lodash/get';
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Card, Label } from 'semantic-ui-react';

export class SeriesCard extends Component {
  renderImage = () => {
    const { data, volume } = this.props;
    const image = (
      <LiteratureCover
        size="small"
        url={_get(data, 'metadata.cover_metadata.urls.medium')}
      />
    );

    if (volume) {
      return (
        <div className="search-result-image">
          <Label floating color="black">
            Volume {volume}
          </Label>
          {image}
        </div>
      );
    }

    return image;
  };

  render() {
    const { data } = this.props;
    return (
      <Card
        link
        centered
        className="fs-book-card"
        onClick={() =>
          goTo(FrontSiteRoutes.seriesDetailsFor(data.metadata.pid))
        }
        data-test={data.metadata.pid}
      >
        <Card.Meta className="discrete">Series</Card.Meta>
        {this.renderImage()}
        <Card.Content>
          <Card.Header>
            <LiteratureTitle title={data.metadata.title} />
          </Card.Header>
          <Card.Meta>
            <div>
              <SeriesAuthors authors={data.metadata.authors} />
            </div>
            {data.metadata.edition && (
              <div>Edition {data.metadata.edition}</div>
            )}
            {data.metadata.publisher && (
              <div>Publisher {data.metadata.publisher}</div>
            )}
          </Card.Meta>
        </Card.Content>
      </Card>
    );
  }
}

SeriesCard.propTypes = {
  data: PropTypes.object.isRequired,
  volume: PropTypes.string,
};

SeriesCard.defaultProps = {
  volume: null,
};
