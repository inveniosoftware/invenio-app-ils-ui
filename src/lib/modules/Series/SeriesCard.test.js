import React from 'react';
import { mount } from 'enzyme';
import testData from '@testData/series.json';
import { FrontSiteRoutes } from '@routes/urls';
import history from '@history';
import { SeriesCard } from './SeriesCard';

describe('SeriesCard tests', () => {
  let component;
  afterEach(() => {
    if (component) {
      component.unmount();
    }
  });

  const seriesData = {
    metadata: {
      ...testData[0],
      title: 'Lorem',
      edition: '12',
      authors: ['Author1', 'Author2'],
      mode_of_issuance: 'SERIAL',
    },
  };

  it('should render the SeriesCard', () => {
    component = mount(<SeriesCard data={seriesData} />);
    expect(component).toMatchSnapshot();

    const rows = component
      .find('SeriesCard')
      .find('Card')
      .filterWhere(
        (element) => element.prop('data-test') === seriesData.metadata.pid
      );
    expect(rows).toHaveLength(1);
  });

  it('should go to book details when clicking on a book', () => {
    const mockedHistoryPush = jest.fn();
    history.push = mockedHistoryPush;
    component = mount(<SeriesCard data={seriesData} />);
    expect(component).toMatchSnapshot();
    const card = component.find('SeriesCard');
    card.simulate('click');
    const expectedParam = FrontSiteRoutes.seriesDetailsFor(
      seriesData.metadata.pid
    );
    expect(mockedHistoryPush).toHaveBeenCalledWith(expectedParam, {});
  });
});
