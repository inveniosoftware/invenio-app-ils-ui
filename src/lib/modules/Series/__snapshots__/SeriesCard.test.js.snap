// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeriesCard tests should go to book details when clicking on a book 1`] = `
<SeriesCard
  data={
    Object {
      "metadata": Object {
        "$schema": "https://127.0.0.1:5000/schemas/series/series-v1.0.0.json",
        "abstract": "This is a multipart monograph",
        "authors": Array [
          "Author1",
          "Author2",
        ],
        "edition": "12",
        "mode_of_issuance": "SERIAL",
        "pid": "serid-1",
        "title": "Lorem",
      },
    }
  }
  volume={null}
>
  <Card
    centered={true}
    className="fs-book-card"
    data-test="serid-1"
    link={true}
    onClick={[Function]}
  >
    <a
      className="ui centered link card fs-book-card"
      data-test="serid-1"
      onClick={[Function]}
    >
      <CardMeta
        className="discrete"
      >
        <div
          className="meta discrete"
        >
          Series
        </div>
      </CardMeta>
      <Overridable(LiteratureCover)
        size="small"
      >
        <LiteratureCover
          asItem={false}
          isLoading={false}
          isRestricted={false}
          linkTo={null}
          size="small"
          url={null}
        >
          <Overridable
            asItem={false}
            id="LiteratureCover.placeholder"
            isLoading={false}
            isRestricted={false}
            linkTo={null}
            size="small"
            url={null}
          >
            <Placeholder>
              <div
                className="ui placeholder"
              >
                <PlaceholderImage
                  square={true}
                >
                  <div
                    className="square image"
                  />
                </PlaceholderImage>
              </div>
            </Placeholder>
          </Overridable>
        </LiteratureCover>
      </Overridable(LiteratureCover)>
      <CardContent>
        <div
          className="content"
        >
          <CardHeader>
            <div
              className="header"
            >
              Lorem
            </div>
          </CardHeader>
          <CardMeta>
            <div
              className="meta"
            >
              <div>
                <SeriesAuthors
                  authors={
                    Array [
                      "Author1",
                      "Author2",
                    ]
                  }
                >
                  <div
                    className="document-authors-list-wrapper"
                  >
                    <SeparatedList
                      className="document-authors-list"
                      itemProps={Object {}}
                      items={
                        Array [
                          "Author1",
                          "Author2",
                        ]
                      }
                      prefix={null}
                      separator="; "
                      suffix={null}
                    >
                      <List
                        className="document-authors-list"
                        horizontal={true}
                      >
                        <div
                          className="ui horizontal list document-authors-list"
                          role="list"
                        >
                          <ListItem
                            key="Author1"
                          >
                            <div
                              className="item"
                              onClick={[Function]}
                              role="listitem"
                            >
                              Author1
                              ; 
                            </div>
                          </ListItem>
                          <ListItem
                            key="Author2"
                          >
                            <div
                              className="item"
                              onClick={[Function]}
                              role="listitem"
                            >
                              Author2
                            </div>
                          </ListItem>
                        </div>
                      </List>
                    </SeparatedList>
                  </div>
                </SeriesAuthors>
              </div>
              <div>
                Edition 
                12
              </div>
            </div>
          </CardMeta>
        </div>
      </CardContent>
    </a>
  </Card>
</SeriesCard>
`;

exports[`SeriesCard tests should render the SeriesCard 1`] = `
<SeriesCard
  data={
    Object {
      "metadata": Object {
        "$schema": "https://127.0.0.1:5000/schemas/series/series-v1.0.0.json",
        "abstract": "This is a multipart monograph",
        "authors": Array [
          "Author1",
          "Author2",
        ],
        "edition": "12",
        "mode_of_issuance": "SERIAL",
        "pid": "serid-1",
        "title": "Lorem",
      },
    }
  }
  volume={null}
>
  <Card
    centered={true}
    className="fs-book-card"
    data-test="serid-1"
    link={true}
    onClick={[Function]}
  >
    <a
      className="ui centered link card fs-book-card"
      data-test="serid-1"
      onClick={[Function]}
    >
      <CardMeta
        className="discrete"
      >
        <div
          className="meta discrete"
        >
          Series
        </div>
      </CardMeta>
      <Overridable(LiteratureCover)
        size="small"
      >
        <LiteratureCover
          asItem={false}
          isLoading={false}
          isRestricted={false}
          linkTo={null}
          size="small"
          url={null}
        >
          <Overridable
            asItem={false}
            id="LiteratureCover.placeholder"
            isLoading={false}
            isRestricted={false}
            linkTo={null}
            size="small"
            url={null}
          >
            <Placeholder>
              <div
                className="ui placeholder"
              >
                <PlaceholderImage
                  square={true}
                >
                  <div
                    className="square image"
                  />
                </PlaceholderImage>
              </div>
            </Placeholder>
          </Overridable>
        </LiteratureCover>
      </Overridable(LiteratureCover)>
      <CardContent>
        <div
          className="content"
        >
          <CardHeader>
            <div
              className="header"
            >
              Lorem
            </div>
          </CardHeader>
          <CardMeta>
            <div
              className="meta"
            >
              <div>
                <SeriesAuthors
                  authors={
                    Array [
                      "Author1",
                      "Author2",
                    ]
                  }
                >
                  <div
                    className="document-authors-list-wrapper"
                  >
                    <SeparatedList
                      className="document-authors-list"
                      itemProps={Object {}}
                      items={
                        Array [
                          "Author1",
                          "Author2",
                        ]
                      }
                      prefix={null}
                      separator="; "
                      suffix={null}
                    >
                      <List
                        className="document-authors-list"
                        horizontal={true}
                      >
                        <div
                          className="ui horizontal list document-authors-list"
                          role="list"
                        >
                          <ListItem
                            key="Author1"
                          >
                            <div
                              className="item"
                              onClick={[Function]}
                              role="listitem"
                            >
                              Author1
                              ; 
                            </div>
                          </ListItem>
                          <ListItem
                            key="Author2"
                          >
                            <div
                              className="item"
                              onClick={[Function]}
                              role="listitem"
                            >
                              Author2
                            </div>
                          </ListItem>
                        </div>
                      </List>
                    </SeparatedList>
                  </div>
                </SeriesAuthors>
              </div>
              <div>
                Edition 
                12
              </div>
            </div>
          </CardMeta>
        </div>
      </CardContent>
    </a>
  </Card>
</SeriesCard>
`;
